storage/namespace.go

```diff
@@ -66,7 +67,8 @@ type dbNamespace struct {
 	blockRetriever block.DatabaseBlockRetriever
 	opts           Options
 	runtimeOptsMgr m3dbruntime.OptionsManager
-	nopts          namespace.Options
+	metadata       namespace.Metadata
+	seriesOpts     series.Options
```

Comment:
This is the compromise I ended up with because `series.Options` need to be aware of retention (they use bufferFuture/bufferPast/blockSize). I could push this down and have each shard create the appropriate series options, but doing it once in the namespace and passing down will have a lot fewer option objects floating around.